src/main/java/org/nypl/findawayclientlibrary/DownloadService.java

```diff
@@ -181,7 +181,7 @@ public void onError(Throwable e) {
         public void onNext(Integer progress) {
           LogHelper.d(TAG, "Got initial download progress ", progress);
 
-          callbackActivity.setDownloadProgress(progress, 0, null);
+          callbackActivity.drawDownloadProgressButton(progress, 0, null);
```

Comment:
It seems wrong to me that a class named `DownloadService` is invoking some sort of draw method. My expectation would be that this class makes no assumptions about the UI.

Reply:
It's invoking that method on the UI activity class.  It has no say in how the activity chooses to draw.  There is a question of what to do, if an activity is destroyed while a download or playback runs.  That's a waffle issue I haven't addressed here.